mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java

```diff
@@ -395,6 +395,25 @@ public void testInsertFromSelectWithProjection() throws IOException {
     HiveIcebergTestUtils.validateData(table, expected, 0);
   }
 
+  @Test
+  public void testInsertFromHiveTableWithSameColumnNames() throws IOException {
+    Assume.assumeTrue("Tez write is not implemented yet", executionEngine.equals("mr"));
+
+    shell.executeStatement(
+        "CREATE TABLE hive_customers(customer_id bigint, first_name string) PARTITIONED BY (last_name string)");
+    shell.executeStatement(
+        "INSERT INTO hive_customers VALUES (0, 'Alice', 'Brown'), (1, 'Bob', 'Green'), (2, 'Trudy', 'Pink')");
```

Comment:
Can we use `testTables.createTable(TestHiveShell shell, String tableName, Schema schema, PartitionSpec spec, FileFormat fileFormat, List<Record> records)` here?

Or it would be even better if we create a version of this method where we do not need to execute a Hive query to insert the data. We do not test the Hive `INSERT INTO` here and adding records directly to the table is faster.

Reply:
I switched to using Iceberg tables to make it simpler. Originally I used Hive tables in the test because with Hive, `SELECT *` does not put the partition column into `hive.io.file.readcolumn.names` which leads to this unexpected failure, while this behaviour regarding partition columns is not present in Iceberg. However, we can reproduce the issue by using an Iceberg source table as well (with the same columns as the target table), just not selecting all of its columns.